roadmap/implementers-guide/src/runtime/scheduler.md

```diff
@@ -1,3 +1,227 @@
+   group to prepare and send a collation.
+2. Collators need to know in advance what collators for which
+   parachains/parathreads they should accept connections and collations from.
+3. Users need to know whether they won a slot or not - seeing assigned cores
+   could be one way of achieving that goal. (Likely not very suitable though).
+
+### Interface to the scheduler
+
+Once we found claims for cores, we need to get them to the scheduler. This
+interface will roughly be:
+
+Give me claims for this claim queue size for this number of cores. For
+simplicity we make the number of cores per assignment provider a configuration
+that can only be changed on session boundaries.
+
+Basically the scheduler needs to be able to provide core assignments for the
+look ahead period required by asynchronous backing. The assignment providers
+(like the parathread one above) need to be able to provide those, given a
+configured number of available cores to the provider.
+
+Assuming we treat cores separately and sequentially (see questions below), an
+assignment provider could return an enum like this:
+
+```rust
+enum Assignment {
+  // Parathreads
+  List(Vec<Claim>),
+  // slot sharing/slow parachains:
+  // Useful for parachains, which want a 12 second block time. - With async
+  backing, all parachains will be of this type in the beginning:
+  Alternating2(Claim, Claim),
+  // 18 second parachains ... cost a third.
+  Alternating3(Claim, Claim, Claim),
+  // For parachains/booster
+  Single(Claim),
+}
+```
+
+per core.
+
+So the interface could look something like this:
+
+```rust
+trait AssignmentProvider {
+  /// Upcoming assignments that are ready from the provider.
+  pub fn lookahead(&self) -> UpcomingAssignments
+  /// Tell provider that an item in the claim queue at the given index has been
+  served (or timed out) - in any case core became free.
+  pub fn pop_assignment(index: AssignmentIndex);
+}
+
+
+```
+
+^^ To initiate a new round `set_num_cores`... needs to be called with a
+`RoundId`, can only be called once per round.
+
+Questions:
+
+Can different cores in a block and cores across blocks be treated the same? E.g.
+if a parathread a parachain has two claims, would it be legal to give it two
+cores in the same block?
+
+Good point brought up by Alex: We assume that everything is working clock worky
+once backed, but what if availability is taking longer? What does happen to the
+assignments?
+
+-> I guess we need to be able to overflow: If some assignments could not be
+served, we need to be able to tell the assignment provider. The assignment
+provider should then consider those left overs in the next round of assignments.
+
+E.g. Either refund/ replace assignments in the next batch.
+
+-> Other questions? Demand driven variation: E.g. demand on parathreads is very
+low, but demand for parachain boost exists - re-assign cores to different
+assignment provider?
```

Comment:
How is parachain boost even supposed to work? In my view, parachain boost is not its own assignment provider. If parachains want a boost, they should have to bid for claims in the parathread queue. Or if they have some way around the auction, they should still just add their claims to the parathread queue and have that assignment provider take care of the execution. Hence, parathreads should get all the leftover cores that are not used by parachains.

In that regard, I'm not even sure we need more than two assigners in general: One for session-static parachains and one for on-demand computation, the parathread one. I can imagine all other assignment provider strategies could be modelled using these two.